File: 0605-can-place-flowers/0605-can-place-flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        flowerbed.insert(flowerbed.begin(),0);
        if (n==0)
            return true;
        flowerbed.push_back(0);
        for (int i=1;i<flowerbed.size()-1;i++) {
            if (flowerbed[i]==0 ) {
            if (flowerbed[i-1] ==0 && flowerbed[i+1] ==0) {
                    flowerbed[i] =1;
                    n-=1;
                    if (n==0)
                        return true;
        }
        }
        }
        return false;
    }
};

File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        if n==0:
            return True
        for i in range(len(flowerbed)):
            if flowerbed[i]==0:
                if (i-1<0 and i+1 >=len(flowerbed)) or (i-1 <0 and flowerbed[i+1] ==0) or  (i+1 >=len(flowerbed) and flowerbed[i-1]==0) or (i-1 >=0 and i+1 <len(flowerbed) and flowerbed[i+1] ==0 and flowerbed[i-1]==0):
                    flowerbed[i]=1
                    n-=1
                    if n==0:
                        break
        if n==0:
            return True
        else:
            return False
                    
        
         